hormonal-hedonists/TLOA/core/game.py

```diff
@@ -0,0 +1,160 @@
+import random
+
+from typing import List, Optional
+
+from TLOA.core.constants import (
+    Actions, AUDIO_PATH, TICK, NUMBER_OF_LANES, LANE_BOUNDS, SHIP_SPAWN_CHANCE,
+    SHIP_SPAWN_RATE, GOLD_SHIP_CHANCE, LIGHT_SOURCE_POS, LIGHT_FOCUS_OFFSET,
+    MIRROR_CANNON_RANGE
+)
+from TLOA.entities import MirrorCannon, BrownShip, GoldenShip, LightRays
+
+from kivy import Logger
+from kivy.clock import Clock, ClockEvent
+from kivy.core.audio import Sound, SoundLoader
+from kivy.event import EventDispatcher
+from kivy.properties import BooleanProperty, BoundedNumericProperty, NumericProperty
+from kivy.vector import Vector
+
+
+class Game(EventDispatcher):
+    running = BooleanProperty(False)
+    score = NumericProperty(0)
+    health = BoundedNumericProperty(100, min=0, max=100,
+                                    errorhandler=lambda x: 0 if x < 0 else 100)
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.pause_game = False
+        self._sound = None
+        self.mirror = MirrorCannon()
+
+        self._island_destroyed: Sound = SoundLoader.load(AUDIO_PATH.format("mirror_moving.wav"))
+        self._mirror_moving: Sound = SoundLoader.load(AUDIO_PATH.format("mirror_moving.wav"))
+        self._ship_destroyed: Sound = SoundLoader.load(AUDIO_PATH.format("ship_explosion.mp3"))
+
+        # Create & initialize Mesh instance for Incident Sun rays
+        self.sun_rays = LightRays(point=LIGHT_SOURCE_POS, surface=self.mirror.mirror_axis)
+
+        # Create & initialize Mesh instance for Reflected Sun rays/Death Rays
+        self.death_rays = LightRays(
+            point=Vector(MIRROR_CANNON_RANGE, LANE_BOUNDS[self.mirror.state][1]),
+            surface=self.mirror.mirror_axis,
+            angle=self.mirror.angle)
+
+        self.ship_lanes: List[List[BrownShip]] = [[] for _ in range(NUMBER_OF_LANES)]
+
+        self._event: Optional[ClockEvent] = None
+        self.register_event_type('on_add_ship')
+        self.register_event_type('on_remove_ship')
+        self.bind(health=self.on_health_change)
+
+    def start(self):
+        Logger.info('Start game')
```

Comment:
I like the use of the Kivy `Logger`; it shows that you're trying to make maximum use of the tools Kivy offers you.